models.py

```diff
@@ -326,23 +299,59 @@ def __str__(self):
             self.proofreader.full_name(),
         )
 
-    def cancel(self):
+    def cancel(self, request):
```

Comment:
I don't think the models should depend on the request, since it locks the models from being used outside of the context of an HTTP request. Instead it should either be an optional argument or retrieved from the global context (ideally we would do this from the event framework instead).